docs-v2/guides/webhooks/webhooks-from-nango.mdx

```diff
@@ -11,30 +11,147 @@ Nango sends webhook notifications to your backend in different cases:
+- `override`: a connection has been re-authorized
+- `refresh`: an OAuth connection's access token has been refreshed
+
+Payload received following a refresh token error: 
+```json
+{
+    "type": "auth",
+    "operation": "refresh",
+    "connectionId": "<your-connection-id>",
+    "authMode": "OAUTH2 | BASIC | API_KEY | ...",
+    "providerConfigKey": "<your-integration-id>",
+    "provider": "<your-provider-key>",
+    "environment": "DEV | PROD | ...",
+    "success": false,
+    "endUser": { 
+        "endUserId": "<your-end-user-id>", 
+        "organizationId": "<your-organization-id>"
+    },
+    "error": {
+        "type": "<string>",
+        "description": "<string>"
+    }
+}
+```
+
+
+<Note>
+Webhooks are only sent for certain connection creation errors. For example, during the OAuth flow, some errors are reported locally in the OAuth modal by the external API. Since Nango does not receive these errors, it cannot trigger a webhook for them.
+</Note>
+
+### Sync webhooks
+
+Sync webhooks are sent when a [sync](/guides/syncs/overview) execution finishes, whether successful or not.
+
+Payload received following a successful sync execution: 
+```json
+{
+    "type": "sync",
+    "connectionId": "<your-connection-id>",
+    "providerConfigKey": "<your-integration-id>",
+    "syncName": "<your-sync-script-name>",
+    "model": "<your-model-name>",
+    "syncType": "INCREMENTAL | INITIAL | WEBHOOK",
+    "success": true,
+    "modifiedAfter": "<timestamp>",
+    "responseResults": {
+        "added": number,
+        "updated": number,
+        "deleted": number
+    }
+}
+```
+
+By default, Nango sends a webhook even if no modified data was detected in the last sync execution (referred as an "empty" sync), but this is configurable in your _Environment Settings_. In case of an empty sync, the `responseResults` would be: 
+```json
+[
+    "added": 0,
+    "updated": 0,
+    "deleted": 0
+  ]
+```
+
+The `syncType` possible values are:
+- `INITIAL`: for the very first execution of a sync.
+- `INCREMENTAL`: for subsequent executions  of a sync.
+- `WEBHOOK`: when an external API webhooks triggers the execution of the sync's `onWebhookPayloadReceived` function.
+
+Payload received following a failed sync execution: 
+```json
+{
+    "type": "sync",
+    "connectionId": "<your-connection-id>",
+    "providerConfigKey": "<your-integration-id>",
+    "syncName": "<your-sync-script-name>",
+    "model": "<your-model-name>",
+    "syncType": "INCREMENTAL | INITIAL | WEBHOOK",
+    "success": true,
+    "error": {
+        "type": "<string>",
+        "description": "<string>"
+    },
+    "startedAt": "<timestamp>",
+    "failedAt": "<timestamp>"
+}
+```
+
+### External webhook forwarding
+
+Nango can forward webhooks received from 3rd-party APIs. See details in this [guide](/guides/webhooks/webhooks-from-apis#forward-external-webhooks-to-your-app).
+
+# Webhook retries
+
+Nango retries (with exponential backoff) webhooks with non-2xx responses.
```

Comment:
Would it make sense to note we retry 7 times (or however many times it is?)

Reply:
Done!